docs/reference/api/storage/LittleFileSystem.md

```diff
@@ -1,11 +1,130 @@
+  '-----'   '----------'
+   | | |
+```
+
+Microcontrollers and flash storage present three challenges for embedded storage: [power loss](#power-loss-resilience), [wear](#wear-leveling) and [limited RAM and ROM](#bounded-ram-and-rom). The LittleFileSystem provides a solution to all three of these problems.
+
+### Power loss resilience
+
+Embedded systems are usually without a shutdown routine and the notable lack of a user interface for recovery. With a file system that is not resilient to power loss, you rely on luck to not end up with a corrupted file system. The combination of persistent storage and unpredictable power loss creates bugs that are difficult to notice and that ruin the experience of unlucky users.
+
+The LittleFileSystem has a structure that is resilient to power loss. It uses checksums to limit the assumptions of how the physical storage reacts to power loss. The LittleFileSystem provides copy-on-write guarantees and keeps the storage on disk in a valid state.
+
+### Wear leveling
+
+Flash storage presents its own challenge: wear. Flash is a destructive form of storage, and continuously rewriting data to a block causes that block to wear out and become unwritable. File systems that don't account for wear can quickly burn through the blocks that store frequently updated metadata and result in the premature death of the system.
+
+We accounted for wear when we built the LittleFileSyste, and the underlying structure of the file system reactively mutates as the underlying storage develops errors throughout its lifetime. This results in a form of dynamic wear-leveling that extends the lifetime of the physical storage proportionally to the size of storage. With the LittleFileSystem, you can extend the lifetime of storage by increasing the size of storage, which is cheaper than upgrading the erase cycles on storage.
+
+### Bounded RAM and ROM
+
+The scale of resources available to microcontrollers is foreign to the types of systems file systems normally target. A common trend of embedded Linux file systems is RAM usage that scales linearly with the size of storage, which makes rationalizing RAM usage in a system difficult.
+
+The LittleFileSystem works with a limited amount of memory. The LittleFileSystem avoids recursion and limits dynamic memory to configurable buffers. At no point during operation does the LittleFileSystem store an entire storage block in RAM. The result
+is small RAM usage that is independent of the geometry of the underlying storage.
+
+---
+
+The "little" in the LittleFileSystem comes from the focus on both keeping resource usage low and keeping the scope self-contained. Aside from the three targeted issues above, there is a heavy restriction against bloat in this software module. Instead, additional features are pushed separate layers in the powerful [BlockDevice API](TODO LINK ME) that drives the Mbed OS storage stack. This gives Mbed OS a tool for remaining flexible as technology used by IoT devices develops.
```

Comment:
Clean up(TODO LINK ME)

Reply:
Thanks for marking these, we were leaving the unknown links for later in case we found a good candidate